wordpress-mobile/WordPress-Android
# [Jetpack Plugin Install] Reuse the tweaked UI for remote Jetpack install

Fixes #18045 

Update the existing "Remote Install" Flow to use this same updated flow (made with Compose). This flow is triggered when the user clicks a Jetpack feature (such as Stats) without having the Jetpack plugin installed on their self-hosted site.

## Demo
<details>
<summary>Success Flow</summary>

https://user-images.githubusercontent.com/5091503/224807924-522ddb50-b899-43c2-9751-f70aa5e78610.mov

</details>

<details>
<summary>Error Flow</summary>

https://user-images.githubusercontent.com/5091503/224808004-96729095-7943-4961-8ead-d1c7d9a5b40c.mp4

</details>

## To test
Set up a self-hosted site that has only no Jetpack plugins at all.

1. Install Jetpack OR WordPress app
2. Login using the site credentials
3. Select the self-hosted site from the site list
4. Tap a Jetpack feature, such as **Stats**
5. **Testing the error state**
    1. Turn on Airplane mode
    2. Start the Plugin Install Flow
    3. **Verify** the UI is updated according to expectations
6. **Testing the install flow**
    1. Close the Plugin Install flow
    2. Turn off Airplane mode
    3. Tap a Jetpack feature, such as **Stats**
    4. Start the Plugin Install Flow
    5. **Verify** the UI is updated according to expectations
    6. **Verify** everything worked and the Jetpack plugin was installed

## Regression Notes
1. Potential unintended areas of impact
Changes in installation behavior.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual test (it is working exactly as it was before).

3. What automated tests I added (or what prevented me from doing so)
Updated existing tests.

PR submission checklist:
